server_addon/houdini/client/ayon_houdini/startup/scripts/OnCreated.py

```diff
@@ -0,0 +1,71 @@
+# Any code here will run on any node being created in Houdini
+# As such, preferably the code here should run fast to avoid slowing down node
+# creation. Note: It does not trigger on existing nodes for scene open nor on
+# node copy-pasting.
+from ayon_core.lib import env_value_to_bool
+
+
+def make_publishable(node, product_type):
+    # TODO: Can we make this imprinting much faster? Unfortunately
+    #  CreateContext initialization is very slow.
+    from ayon_core.pipeline import registered_host
+    from ayon_core.pipeline.create import CreateContext
+
+    host = registered_host()
+    context = CreateContext(host)
+
+    variant = node.name()
+
+    # Apply the instance creation to the node
+    context.create(
+        creator_identifier="io.ayon.creators.houdini.publish",
+        variant=variant,
+        pre_create_data={
+            "productType": product_type,
+            "node": node
+        }
+    )
```

Comment:
@iLLiCiTiT do you have a more optimal way to just create a new instance? The `CreateContext` initialization takes a bit of time because it comes with settings queries, instances collections, etc. but what I in essence want to do is just create an instance and don't care about the rest of the context. Is there an approach where that can easily do less and thus have less overhead?

Reply:
Bonus question: if I wanted this on create callback to access something from settings? What's the easiest way to try and remove overhead from separate calls. Any simple time based caching I could do?